test/studies/shootout/meteor/kbrady/meteor-parallel-alt.chpl

```diff
@@ -367,16 +366,18 @@ proc searchLinear(in board, in pos, in used, in placed, currentSolution) {
   var count, evenRows, oddRows, leftBorder, rightBorder,
       s1, s2, s3, s4, s5, s6, s7, s8: int;
 
-  // Lock atomic lock - this is what sync variables do under the hood
-  proc lock() { while l.testAndSet() != false do chpl_task_yield(); }
 
-  // Unlock atomic lock
-  proc unlock() { l.write(false); }
 
   if placed == numPieces {
-    lock();
+
+    // Wait for unlocked state, then lock it, while this task records solution
+    l.waitFor(false);
```

Comment:
That's a little surprising to me. I wonder if the story would be any different in a native qthread sync var world. 

Do you know how much of a perf difference there is?